src/libponyrt/actor/actor.c

```diff
@@ -230,10 +233,55 @@ static void try_gc(pony_ctx_t* ctx, pony_actor_t* actor)
   DTRACE1(GC_END, (uintptr_t)ctx->scheduler);
 }
 
-bool ponyint_actor_run(pony_ctx_t* ctx, pony_actor_t* actor, size_t batch)
+// return true if mute occurs
+static bool maybe_mute(pony_actor_t* actor)
+{
+  // if we become muted as a result of handling a message, bail out now.
+  // we aren't set to "muted" at this point. setting to muted during a
+  // a behavior can lead to race conditions that might result in a
+  // deadlock.
+  // Given that actor's are not run when they are muted, then when we
+  // started out batch, actor->muted would have been 0. If any of our
+  // message sends would result in the actor being muted, that value will
+  // have changed to greater than 0.
+  //
+  // We will then set the actor to "muted". Once set, any actor sending
+  // a message to it will be also be muted unless said sender is marked
+  // as overloaded.
+  //
+  // The key points here is that:
+  //   1. We can't set the actor to "muted" until after its finished running
+  //   a behavior.
+  //   2. We should bail out from running the actor and return false so that
+  //   it won't be rescheduled.
+  if(atomic_load_explicit(&actor->muted, memory_order_relaxed) > 0)
+  {
+    ponyint_mute_actor(actor);
+    return true;
+  }
+
+  return false;
+}
+
+static bool batch_limit_reached(pony_actor_t* actor, bool polling)
+{
+  if(!has_flag(actor, FLAG_OVERLOADED) && !polling)
+  {
+    // If we hit our batch size, consider this actor to be overloaded
+    // only if we're not polling from C code.
+    // Overloaded actors are allowed to send to other overloaded actors
+    // and to muted actors without being muted themselves.
+    ponyint_actor_setoverloaded(actor);
+  }
+
+  return !has_flag(actor, FLAG_UNSCHEDULED);
+}
+
+bool ponyint_actor_run(pony_ctx_t* ctx, pony_actor_t* actor, bool polling)
 {
   pony_assert(!ponyint_is_muted(actor));
   ctx->current = actor;
+  size_t batch = PONY_ACTOR_DEFAULT_BATCH;
```

Comment:
is this checking now to see if polling is true and changing to 1?

Reply:
I don't understand the reference to `this` in your question. The line you're commented on is not implementing the logic you're asking about. The check for polling happens elsewhere (see my response to your other question).